dds/DCPS/Serializer.cpp

```diff
@@ -575,24 +575,35 @@ bool Serializer::read_parameter_id(unsigned& id, size_t& size, bool& must_unders
     must_understand = emheader & emheader_must_understand;
 
     // Get Size
-    // TODO(iguessthislldo) LC
     const unsigned short lc = (emheader >> 28) & 0x7;
-    if (lc < 4) {
-      size = 1 << lc;
+    if (lc == 0) {
+      size = 1;
+    } else if (lc == 1) {
+      size = 2;
+    } else if (lc == 2) {
+      size = 4;
+    } else if (lc == 3) {
+      size = 8;
```

Comment:
```suggestion
    if (lc < 4) {
      size = 1 << lc;
```

I think it's fine to leave this how it was.